libnmstate/nm/lldp.py

```diff
@@ -0,0 +1,339 @@
+#
+# Copyright (c) 2020 Red Hat, Inc.
+#
+# This file is part of nmstate
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License as published by
+# the Free Software Foundation, either version 2.1 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with this program. If not, see <https://www.gnu.org/licenses/>.
+#
+
+from libnmstate.nm import connection
+from libnmstate.schema import LLDP
+
+from .common import NM
+
+
+NM_VLAN_ID_KEY = "vid"
+NM_VLAN_NAME_KEY = "name"
+NM_MACPHY_AUTONEG_KEY = "autoneg"
+NM_MACPHY_PMD_AUTONEG_KEY = "pmd-autoneg-cap"
+NM_MACPHY_MAU_TYPE_KEY = "operational-mau-type"
+NM_PPVLAN_ID_KEY = "ppvid"
+NM_PPVLAN_FLAGS_KEY = "flags"
+NM_MANAGEMENT_ADDR_KEY = "address"
+NM_MANAGEMENT_ADDR_TYPE_KEY = "address-subtype"
+NM_MANAGEMENT_ADDR_IFACE_NUMBER_KEY = "interface-number"
+NM_MANAGEMENT_ADDR_IFACE_NUMBER_TYPE_KEY = "interface-number-subtype"
+NM_MANAGEMENT_ADDR_TYPE_IPV4 = 1
+NM_INTERFACE_TYPE_IFINDEX = 2
+NM_INTERFACE_TYPE_SYSTEM_PORT = 3
+NM_LLDP_STATUS_DEFAULT = -1
+
+
+LLDP_CAP_NAMES = {
+    0b1: LLDP.Neighbors.SYSTEM_CAPABILITY_OTHER,
+    0b10: LLDP.Neighbors.SYSTEM_CAPABILITY_REPEATER,
+    0b100: LLDP.Neighbors.SYSTEM_CAPABILITY_MAC_BRIDGE,
+    0b1000: LLDP.Neighbors.SYSTEM_CAPABILITY_WLAN_AC,
+    0b10000: LLDP.Neighbors.SYSTEM_CAPABILITY_ROUTER,
```

Comment:
```suggestion
    0b1_0000: LLDP.Neighbors.SYSTEM_CAPABILITY_ROUTER,
```

By separating the nibbles, it is easier to read the value because counting the zeroes is not so nice.

Reply:
Sorry, I am confused by that. How does this work? Thanks!

Reply:
Sorry I am confused by that, so do you propose to split the last bit from the other zeroes? Sorry if I am missing something, thanks!

Reply:
Just add an underscore after 4 binary digits like `0b1_0000` or `0b10_0000` `0b1_0000_0000` it is easier to read than `0b100000000`. It is like adding commas in English numbers 1,000,000.

Reply:
Thanks for explaining :-)

Reply:
Done, thanks!